wpcom-legacy-redirector.php

```diff
@@ -455,6 +455,9 @@ public static function vip_legacy_redirect_parent_id( $post ) {
 				return true;
 			}
 		} else {
+			// Ensure we have a post object.
+			$post = get_post( $post );
```

Comment:
Before calling `get_post()` unnecessarily, we should check that it's not an object.